### PR TITLE
Add call joiing and leaving tests.

### PR DESCRIPTION
**Pull request by richardfazzio**

### CODE REVIEW

File: test/browser/call-joining.js [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+            peerUser2.exec('joinConference', call.callId, {audio: true, video: false}),
+            expectEvents(client, [{
+                type: 'callStatus',
+                predicate: evt => evt.reason === 'callStateChanged' && evt.call.state === Circuit.Enums.CallStateName.Active
+            }, {
+                type: 'callStatus',
+                predicate: evt => evt.reason === 'participantJoined'
+            }])
+        ]);
+        await sleep(5000);
```
Comment (rogeru) [inline test/browser/call-joining.js]: I don't think we need this. We already verify that the event with participantJoined is received. We can add the "call.participants.some..." check in the predicate.

No need to wait and find the call.

File: test/browser/call-joining.js [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+            }])
+        ]);
+        await sleep(3000);
+        call = await client.findCall(call.callId);
+        assert(!call.participants.some(user => user.userId === peerUser1.userId));
+    });
+
+    it('should end call', async () => {
+        await client.endCall(call.callId);
+        await sleep(6000);
```
Comment (rogeru) [inline test/browser/call-joining.js]: We really need 6s?

File: test/browser/directcall.js [hunk 3]
```diff
@@ -5,15 +5,14 @@ import { expectEvents, updateRemoteVideos, sleep, logEvents } from '../helper.js
...
-    this.timeout(15000);
-
-    let client;
-    let peerUser;
-    let call;
-
+    this.timeout(60000);
+    
     before(async function() {
+        await sleep(15000);
```
Comment (rogeru) [inline test/browser/directcall.js]: Do we need this?

File: test/browser/directcall.js [hunk 4]
```diff
@@ -42,11 +41,12 @@ describe('Outgoing direct call', async function() {
         call = res[0];
         assert(call.callId);
         document.querySelector('#localVideo').src = call.localVideoUrl;
-    }).timeout(60000);
+    });
 
     it('should get callStatus event for remoteStreamUpdated and state Active upon peer answering', async () => {
+        await sleep(5000);
```
Comment (rogeru) [inline test/browser/directcall.js]: Do we need this?

File: test/browser/call-joining.js [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+            expectEvents(client, [{
+                type: 'callStatus',
+                predicate: evt => evt.reason === 'participantRemoved' && evt.call.callId === call.callId && !evt.call.participants.some(user => user.userId === peerUser2.userId)
+            }])
+        ]);
+    });
+    
+    it('should add participant to call with addParticipant and check that they are NOT a guest', async () => {
+        await client.addParticipant(call.convId, [peerUser1.userId], true);
+        await sleep(3000);
```
Comment (rogeru) [inline test/browser/call-joining.js]: For any sleep we need, can you please add a comment why its needed?

File: test/browser/directcall.js (hunk 4 above)
Comment (richardfazzio) [reply test/browser/directcall.js]: That is a part of an old patch.

File: test/browser/call-joining.js (hunk 5 above)
Comment (richardfazzio) [reply test/browser/call-joining.js]: done.

File: test/browser/directcall.js (hunk 3 above)
Comment (richardfazzio) [reply test/browser/directcall.js]: That is a part of an old patch.

**Pull request merged**